src/trunk.c

```diff
@@ -8811,75 +8824,135 @@ trunk_branch_count_num_tuples(trunk_handle *spl,
 {
-   platform_log_handle *log_handle = (platform_log_handle *)arg;
+   trunk_node_stats    *node_stats = arg;
+   platform_log_handle *log_handle = node_stats->log_handle;
    trunk_node           node;
    trunk_node_get(spl->cc, addr, &node);
 
+   // clang-format off
+   const char *dashes = "------------------------------------------------------------------";
+   const char *dashes_long = "-------------------------------------------------------------------------------------------------";
+   // clang-format on
+
+   platform_log(log_handle, "%s\n", dashes);
    platform_log(
-      log_handle,
-      "------------------------------------------------------------------\n");
-   platform_log(
-      log_handle, "| node %12lu height %u\n", addr, trunk_height(&node));
-   platform_log(
-      log_handle,
-      "------------------------------------------------------------------\n");
+      log_handle, "| Trunk node=%lu, height=%u\n", addr, trunk_height(&node));
+   platform_log(log_handle, "%s\n", dashes);
+
    uint16 num_pivot_keys = trunk_num_pivot_keys(spl, &node);
-   platform_log(log_handle, "| pivots:\n");
+   platform_log(log_handle, "| pivots: %u\n", num_pivot_keys);
    for (uint16 pivot_no = 0; pivot_no < num_pivot_keys; pivot_no++) {
       char key_str[128];
       trunk_key_to_string(spl, trunk_get_pivot(spl, &node, pivot_no), key_str);
       platform_log(log_handle, "| %u: %s\n", pivot_no, key_str);
    }
 
+   platform_log(log_handle, "%s\n", dashes_long);
+
    // clang-format off
    platform_log(log_handle,
-         "-----------------------------------------------------------------------------------\n");
-   platform_log(log_handle,
-         "| branch |     addr     |  num tuples  | num kv bytes |    space    |  space amp  |\n");
-   platform_log(log_handle,
-         "-----------------------------------------------------------------------------------\n");
+         "    | branch |     addr     |  num tuples  |       num kv bytes        |    space    |space amp |\n");
    // clang-format on
+
+   platform_log(log_handle, "%s\n", dashes_long);
+
    uint16 start_branch = trunk_start_branch(spl, &node);
    uint16 end_branch   = trunk_end_branch(spl, &node);
+   uint16 num_branches = 0;
+   uint64 sum_ntuples  = 0;
+   uint64 sum_nkvbytes = 0;
+
+   size_t nkvbytes = 0;
    for (uint16 branch_no = start_branch; branch_no != end_branch;
         branch_no        = trunk_add_branch_number(spl, branch_no, 1))
    {
       uint64 addr = trunk_get_branch(spl, &node, branch_no)->root_addr;
       uint64 num_tuples_in_branch;
-      uint64 kv_bytes_in_branch;
+      uint64 kv_bytes_in_branch = 0;
+      num_branches++;
+
       trunk_branch_count_num_tuples(
          spl, &node, branch_no, &num_tuples_in_branch, &kv_bytes_in_branch);
-      uint64 kib_in_branch = 0;
+
+      nkvbytes = kv_bytes_in_branch;
+
+      sum_ntuples += num_tuples_in_branch;
+      sum_nkvbytes += kv_bytes_in_branch;
+
+      uint64 kib_in_branch = 1;
```

Comment:
This is where a "fix" was sort-of attempted to resolve the main item raised under issue #156.

This changes old L8855 from value 0 to 1. Otherwise, kib_in_branch computed below on L8900, below, would always come out as 0.

This change does not really fix something; just makes some things come out as non-zero.

I'm not even sure if this is needed, and / or should be retained. Comments welcome during review discussion.

I think this area is where some more code / logic is needed as indicated by call to non-existent function trunk_branch_extent_count().

Reply:
Without actually counting the extents in the branch, which we're not doing any more, we cannot compute a space amp.  And kib_in_branch is superceded by nkvbytes.

So I think we can just kill 8896-8901 and 8911-8912.